lib/lang/type-of.js

```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+/**
+ * @module lib/lang/typeof
+ **/
+
+const TYPE_EXP = /^\[object (.*)\]$/
+const toString = Object.prototype.toString
+
+module.exports = function typeOf(value) {
+  if (value === null) return 'null'
+  if (value === undefined) return 'undefined'
+  const parts = TYPE_EXP.exec(toString.call(value))
+  return parts[1].toLowerCase()
+}
+
+
+/**
```

Comment:
shouldn't we have this docstring above the function?

Reply:
We can, yeah.  Having them at the bottom keeps the clutter out of the code, and is a pattern we used at my former company.  Here, it doesn't matter.  I'll move it.

Reply:
moved